software/spmd/bsg_cuda_lite_runtime/conv1d/kernel_conv1d.c

```diff
@@ -0,0 +1,51 @@
+
+/*!
+  Takes a vector A of length N and a 1D filter of size F, padding size P, and stride S.
+  Performs 1D convolution of A with the filter and stores the result in a vector B
+  of size M = 1 + (N - F + 2P) / S.
+*/
+
+#include "bsg_manycore.h"
+#include "bsg_set_tile_x_y.h"
+#include "math.h"
+
+#define BSG_TILE_GROUP_X_DIM bsg_tiles_X
+#define BSG_TILE_GROUP_Y_DIM bsg_tiles_Y
+
+#include "bsg_tile_group_barrier.h"
+
+INIT_TILE_GROUP_BARRIER(r_barrier, c_barrier, 0, bsg_tiles_X - 1, 0, bsg_tiles_Y - 1);
+
+__attribute__((noinline))
+int kernel_conv1d(const float *A,
+                  const int N,
+                  const float *filter,
+                  const int F,
+                  const int P,
+                  float *B,
+                  const int S,
+                  const int block_size)
+{
+        const int tile_group_idx = __bsg_grid_dim_x * __bsg_tile_group_id_y + __bsg_tile_group_id_x;
+        const int start = tile_group_idx * block_size;
+        const int end = start + block_size;
+        const int M = 1 + (N - F + 2 * P) / S;
+        const int num_cores = bsg_tiles_X * bsg_tiles_Y;
+        for(int i = start + __bsg_id; i < end; i += num_cores)
```

Comment:
A simple comment here, explaining how this works would be awesome.

Reply:
Done!